sway/tree/workspace.c

```diff
@@ -657,11 +657,18 @@ static void set_workspace(struct sway_container *container, void *data) {
 	container->workspace = container->parent->workspace;
 }
 
+static void workspace_reset_layout(struct sway_workspace *workspace) {
+	if (!workspace->tiling->length) {
+		workspace->layout = output_get_default_layout(workspace->output);
```

Comment:
Hooked this into adding a tiling child to the workspace. This means that when we've emptied the workspace, we might still have its representation be `H[]` until the next child is added, when it'll turn back into `S[app]`. This is not ideal, but as far as I can tell we don't have a reliable "workspace has emptied" or "workspace container deleted" signal.

Reply:
`container_detach` may be a better spot? I *think* that every tiling container get removed using that.

Reply:
I was worried that some places may be poking into `workspace->tiling` directly, but it looks like everywhere that ends up calling `workspace_update_representation` does use `container_detach` before that, so I've moved that logic there.